experimental/framework/fluid-static/src/containerCode.ts

```diff
@@ -9,59 +9,153 @@ import {
     DataObjectFactory,
     defaultRouteRequestHandler,
 } from "@fluidframework/aqueduct";
+import { IEvent, IEventProvider } from "@fluidframework/common-definitions";
+import { IAudience } from "@fluidframework/container-definitions";
+import { Container } from "@fluidframework/container-loader";
 import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
-import { IFluidDataStoreFactory, NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
 import { IFluidHandle, IFluidLoadable } from "@fluidframework/core-interfaces";
+import { IChannelFactory } from "@fluidframework/datastore-definitions";
+import { NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
 import { requestFluidObject } from "@fluidframework/runtime-utils";
 
-export type IdToDataObjectCollection = Record<string, IFluidStaticDataObjectClass>;
+import {
+    DataObjectClass,
+    LoadableObjectClass,
+    LoadableObjectClassRecord,
+    LoadableObjectRecord,
+    SharedObjectClass,
+} from "./types";
+import { isDataObjectClass, isSharedObjectClass } from "./utils";
 
-export interface IFluidStaticDataObjectClass {
-    readonly factory: IFluidDataStoreFactory;
+export interface IFluidContainerEvents extends IEvent {
+    (event: "connected", listener: (clientId: string) => void): void;
+}
+
+/**
+ * FluidContainer defines the interface that the developer will use to interact with Fluid.
+ */
+export interface FluidContainer
+    extends Pick<Container, "audience" | "clientId">, IEventProvider<IFluidContainerEvents> {
+    /**
+     * The initialObjects defined in the container config
+     */
+    readonly initialObjects: LoadableObjectRecord;
```

Comment:
Would we also want to add the dynamicObjectTypes to the FluidContainer interface? That way a developer would be able to reference the FluidContainer itself to know what types it supported, and be able to call create based on one of those types?

Reply:
Because this is a static code loading case the developer is responsible for providing the dynamic types. This means they already know what types are available on the container. So I don't think this is needed. We are also not using ids to define types but the types themselves so I'm unsure what this api would look like.